lib/authorize.js

```diff
@@ -0,0 +1,55 @@
+import { getLogger } from 'appium-logger';
+import { exec } from 'teen_process';
+import { getPath } from 'appium-xcode';
+import path from 'path';
+
+export async function authorize(insecure) {
+
+  let logger = getLogger('authorize');
+  let xcodeDir;
+  let user;
+
+  try { 
+    // enable developer tools
+    logger.info('Enabling DevToolsSecurity');
+    await exec('DevToolsSecurity --enable');
+
+    // update security db -- removes authorization prompt
+    logger.info(`Updating security db for ${insecure ? 'insecure' :
+        'developer'} access`);
+    let cmd = (`security authorizationdb write system.privilege.taskport ${insecure ? 'allow' : 'is-developer'}`);
+    await exec(cmd);
+    logger.info('Granting access to built-in simulator apps');
+
+    if (!process.env.HOME) {
+      logger.info(new Error('Could not determine your $HOME'));
+    } else { 
+      user = /\/([^\/]+)$/.exec(process.env.HOME)[1];
+    }
+    
+    xcodeDir =  await getPath(); 
+    logger.info(xcodeDir);
+  } catch (e) {  
+    logger.errorAndThrow(e);
+  }
+
+  try{ 
+    // change permission
+    let glob = await path.resolve(xcodeDir, 'Platforms/iPhoneSimulator.platform/' +
```

Comment:
yes, we only need to `await` functions that have been decorated with `async` or are promises, and path.resolve is neither (it's a normal sync js function)